pkgs/tools/misc/btrfs-assistant/default.nix

```diff
@@ -0,0 +1,91 @@
+{ lib
+, stdenv
+, fetchFromGitLab
+, bash
+, btrfs-progs
+, btrfs-snap
+, cmake
+, coreutils
+, git
+, pkg-config
+, qtbase
+, qtsvg
+, qttools
+, snapper
+, util-linux
+, wrapQtAppsHook
+, enableSnapper ? false
```

Comment:
Just to know, why not enabling it as default? I believe that it should be enabled except if there is some drawback.

Reply:
I originally left it always enabled, but thought since it was an optional dependency in the actual program it could be optional to bundle the additional software in the derivation. Snapper is used on Btrfs filesystems to take a snapshot at regular intervals of the filesystem to allow reverting any changes to a prior point in history, which also seemed less needed in NixOS... but I suppose I was close minded in thinking of it only being used there instead of regular Linux distros using Nix.

Reply:
There is also the issue that snapper (according to you) works on filesystem block level and for Btrfs only, whereas Nix/NixOS works at the filesystem file level.

Different tools for different purposes, despite similar.

Reply:
Yeah, true. I went through and tested setting up snapper because I remembered a few use cases I had for it before.